### DISCUSSION TITLE
Which module should be used for creep calculation under the generalized plane strain assumption?

### DISCUSSION THREAD

**Discussion by 09liu** [idaholab/moose#24928] [Q&A Modules: Solid mechanics]
@GiudGiud Which module should be used for creep calculation under the generalized plane strain assumption?

**Comment by GiudGiud** [comment]
@recuero

**Reply by recuero** [reply]
For engineering scale creep using a traditional finite element approach, the module to use is Tensor Mechanics. There are tests that use generalized plane strain theory which you can augment with your creep modeling approach.

FYI: I think you can model creep using the peridynamics module too.

**Reply by 09liu** [reply]
Thank you. I found that PowerLawCreepStressUpdate can perform creep calculations under generalized plane strain assumptions. I modeled this class and constructed new classes, successfully simulating the creep of the material I studied.
